src/python/grpcio/grpc/_channel.py

```diff
@@ -386,32 +404,47 @@ def add_callback(self, callback):
 
     def initial_metadata(self):
         with self._state.condition:
-            while self._state.initial_metadata is None:
-                self._state.condition.wait()
+
+            def _done():
```

Comment:
I'm okay with which ever ways you see fit for these functions. The operator, functions like this, or lambdas.

Reply:
After looking into how CPython is implemented, I determined that this method will be quite a bit faster than either of the other solutions we talked about. And it probably is optimized for people reading the code rather than the person writing the code too.